hudi-aws/src/main/java/org/apache/hudi/aws/transaction/lock/S3StorageLockClient.java

```diff
@@ -267,6 +269,52 @@ private static S3Client createS3Client(Region region, long timeoutSecs, Properti
             .region(region).build();
   }
 
+  @Override
+  public Option<String> readObject(String filePath, boolean checkExistsFirst) {
+    try {
+      // Parse the file path to get bucket and key
+      URI uri = new URI(filePath);
+      String bucket = uri.getHost();
+      String key = uri.getPath().replaceFirst("/", "");
+      
+      if (checkExistsFirst) {
```

Comment:
do u think this is required? does nt it double the cost, was wondering if we just GET the object.

Reply:
It's for performance benefits.

Reply:
ok sg.